parachain/modules/consensus/bnb-pos/verifier/src/lib.rs

```diff
@@ -0,0 +1,102 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+#[warn(unused_imports)]
+#[warn(unused_variables)]
+use anyhow::anyhow;
+use ismp::util::Keccak256;
+use primitives::{parse_extra, CodecHeader, Header};
+use sp_core::{H160, H256};
+use sync_committee_verifier::crypto::verify_aggregate_signature;
+pub mod primitives;
+use alloc::collections::BTreeSet;
+use ark_bls12_381::G1Projective;
+use bls::{point_to_pubkey, pubkey_to_point};
+use core::ops::{Add, AddAssign};
+use sync_committee_primitives::constants::BlsPublicKey;
+use crate::primitives::CodecValidatorInfo;
+
+extern crate alloc;
+
+#[derive(Debug, Clone)]
+pub struct VerificationResult {
+    pub hash: H256,
+    pub header: CodecHeader,
+    pub next_validators: Option<Vec<H160>>,
+}
+
+pub struct ValidatorData {
+    pub address: [u8; 20],
+    pub bls_public_key: [u8; 48],
+}
+
+impl From<CodecValidatorInfo> for ValidatorData {
+    fn from(codec_info: CodecValidatorInfo) -> Self {
+        ValidatorData {
+            address: codec_info.address,
+            bls_public_key: codec_info.bls_public_key,
+        }
+    }
+}
+
+
+pub fn verify_bnb_header<I: Keccak256>(
+    validators: &Vec<ValidatorData>,
+    header: CodecHeader,
+) -> Result<VerificationResult, anyhow::Error> {
+    let rlp_header: Header = (&header).into();
+
+    let parse_extra_data = parse_extra::<I>(&rlp_header.extra_data.0.as_ref())
+        .map_err(|_| anyhow!("could not parse extra data from header"))?;
+
+    let bls_public_keys: Vec<[u8; 48]> = validators
+        .iter()
+        .map(|validator| validator.bls_public_key)
+        .collect();
+
+    let aggregate_public_key = aggregate_public_keys(bls_public_keys.clone())?;
```

Comment:
You are meant to agggregate only the keys at the corresponding positions that have 1 in the validator bit set

Reply:
So, the idea is that, the validators being passed into the method are those that have their corresponding positions in the validator bit set